### ISSUE TITLE
XXD_EXE-NOTFOUND

### ISSUE THREAD

**Issue by posophe** [Ancurio/mkxp#130]
Actually, I can't build mkxp because xxd executable cannot be found by cmake. I defined XXD_EXE manually, but nothing to do, it still doesn't work. Any idea ?

Regards.

**Comment by Ancurio** [comment]
You have to install xxd.

**Comment by posophe** [comment]
I guess so, but I can't find the package it belongs to.

**Comment by Ancurio** [comment]
On Fedora 21 it's part of `vim-common`, not sure about other distros.

**Comment by posophe** [comment]
Got it. Thanks

**Comment by posophe** [comment]
Another little thing. My build fails but I have no output to debug, even with the verbose mode. Does a cmake switch exist to track build error or something like that ?

**Comment by carstene1ns** [comment]
`make VERBOSE=1`, however, if the build fails, you should get an error message anyways.

**Comment by posophe** [comment]
The only thing I have is
In static member function 'static int EventThread::eventFilter(void*, SDL_Event*)':
[  217s] /home/abuild/rpmbuild/BUILD/mkxp/src/eventthread.cpp:497:7: error: 'SDL_RENDER_DEVICE_RESET' was not declared in this scope
[  217s]   case SDL_RENDER_DEVICE_RESET :
[  217s]        ^
It's not that high to produce a failure, isn't it ?

**Comment by sorlok** [comment]
Are you using the very latest (from source) version of SDL?

**Comment by posophe** [comment]
Yes

**Comment by Ancurio** [comment]
Those constants were added in the latest SDL2 release, so your installation is probably out of date. However they're not really in use by mkxp right now so I'll comment out for the time being.

**Comment by posophe** [comment]
Ok. openSUSE Factory provides SDL2 2.0.3, which is the latest version available yet. Anyway, thanks for your help.

**Comment by posophe** [comment]
A last thing. Can I use a switch to make mkxp read configuration file from /etc ?

**Comment by Ancurio** [comment]
> A last thing. Can I use a switch to make mkxp read configuration file from /etc ?

No, that would make little sense, as every configuration is game specific. mkxp is not a samba server :)

> Is iconv an hard dependency and, if yes, is there a way to hardlink to it ?

No, only on OSX. On Linux it will be used when built with the `INI_ENCODING` option (qmake only), and `libguess` will also become a dep. There's no reason to hard link it because it's always provided by glibc on Linux anyay.

> And finally, did you create an icon for mkxp ? I would like to provide a desktop file with my package.

I did make a really basic placeholder one (just so my WM doesn't use the default icon), it's in `asssets/`.

> In fact, as it is defined for each project, it would probably be better to read it from ~/.mkxp

Are you talking about configuration?

**Comment by posophe** [comment]
Yes I am. The configuration file like the one provided as a template is configurable for each game you create with mkxp, isn't it ?

**Comment by Ancurio** [comment]
It's not just configurable for each game, almost all games need their own configuration because they all have different requirements. I don't understand how or why this should be read from `~/.mkxp`. Games assets and scripts are distributed together with mkxp + configuration to form one "game".

**Comment by posophe** [comment]
Because, when you install a package, you need to have a consistent and logical directories tree. In fact, I think a working directory would be the way out. I just asked if there is a way to configure a working directory.

**Comment by Ancurio** [comment]
That's not an issue, since the directory containing the assets must be the current one, and mkxp will read mkxp.conf from the current directory when built with the `WORKDIR_CURRENT` define (as documented in README.md).

**Comment by posophe** [comment]
Yes but it can"t obviously read the conf if the binary is in /usr/bin, it's not possible.

**Comment by Ancurio** [comment]
I just explained to you why it can...

    $ cd ~/my_game
    $ /usr/bin/mkxp

**Comment by posophe** [comment]
Euh ok my bad sorry. Well, it's not really distributable on state but it's the beginning  :)
